### PR TITLE
Context broker notifications

### PR DESCRIPTION
**Pull request by beknazaresenbek**

### CODE REVIEW

File: common/src/main/java/org/fiware/tmforum/common/notification/EventHandler.java [hunk 1]
```diff
@@ -74,44 +76,50 @@ public <T> Mono<Void> handleCreateEvent(T entity) {
...
+        return new TMForumNotification(listenerCallback, event);
+    }
+
     private <T> Map<String, Object> applyFieldsFilter(T entity, List<String> fields) {
         Map<String, Object> entityMap = entityVOMapper.convertEntityToMap(entity);
+
+        if (entity instanceof EntityVO && entityMap.containsKey("additionalProperties") &&
+                entityMap.get("additionalProperties") != null) {
+            entityMap.putAll(entityVOMapper.convertEntityToMap(entityMap.get("additionalProperties")));
+            entityMap.remove("additionalProperties");
```
[the remark this answers is not in the document]
Comment (beknazaresenbek) [reply common/src/main/java/org/fiware/tmforum/common/notification/EventHandler.java:120]: before it's removed, its fields will be moved (or flattened) to the entityMap

[the remark this answers is not in the document]
Comment (pulledtim) [reply common/src/main/java/org/fiware/tmforum/common/notification/EventHandler.java:120]: Yes but is it possible that the fields that are moved/flattened contain a field "additionalProperties" itself?

[the remark this answers is not in the document]
Comment (beknazaresenbek) [reply common/src/main/java/org/fiware/tmforum/common/notification/EventHandler.java:120]: Yes, it's possible. But tmforum filltering by field names selects only top-level fields of an entity (as far as I understood). That's why, lower level additionalProperties will stay unchanged.
image

[the remark this answers is not in the document]
Comment (pulledtim) [reply common/src/main/java/org/fiware/tmforum/common/notification/EventHandler.java:120]: But in the current code, the lowlevel "additionalProperties" field would be deleted right?

```
{
  "testKey": "testValue",
  "additionalProperties": {
    "nestedKey": "nestedValue",
    "additionalProperties": {
      "otherKey": "otherValue"
    }
  }
}
```
would turn into
```
{
  "testKey": "testValue",
  "nestedKey": "nestedValue"
}
```
So maybe we can restructure to:
```
            entityMap.putAll(entityVOMapper.convertEntityToMap(entityMap.remove("additionalProperties")));
``

[the remark this answers is not in the document]
Comment (beknazaresenbek) [reply common/src/main/java/org/fiware/tmforum/common/notification/EventHandler.java:120]: Ok, now I see the problem. Thanks, I'll fix it

[the remark this answers is not in the document]
Comment (beknazaresenbek) [reply common/src/main/java/org/fiware/tmforum/common/notification/EventHandler.java]: Now the method receives an entity object (created by converting entityVO payload) and it doesn't have additionalProperties field anymore

**Review by wistefan — APPROVED** [review]

**Pull request merged**